Movie Challenge/Movie Challenge/Controllers/Home/HomeViewController.swift

```diff
@@ -0,0 +1,139 @@
+            self.bestMovieVotesLabelView.text = "(" + String(movie.vote_count!) + ")"
+            
+            if let releaseDate = movie.release_date{
+                if(!releaseDate.isEmpty){
+                    self.bestMovieYearLabelView.text = String((releaseDate.split(separator: "-").first)!)
+                }
+            }
+            
+            if let runtime = movie.runtime{
+                self.bestMovieRuntimeLabelView.text = String(runtime / 60) + "h" + String(runtime % 60) + "m"
+            } else{
+                self.bestMovieRuntimeLabelView.text = "Duração indefinida"
+            }
+            
+            self.bestMovieCategoryCollectionView.reloadData()
+            let height = self.bestMovieCategoryCollectionView.collectionViewLayout.collectionViewContentSize.height
+            self.bestMovieCategoryCollectionViewHeightConstraint.constant = height
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let movie = bestMovie else { return 0 }
+        guard let genres = movie.genres else { return 0 }
+        
+        return genres.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "bestMovieCategoryCollectionViewCell", for: indexPath) as! BestMovieCategoryCollectionViewCell
+        
+        guard let categoryList = bestMovie.genres else{ return UICollectionViewCell() }
+        cell.setUp(name: categoryList[indexPath.row].name)
+        
+        cell.setCornerRadius()
+        
+        return cell
+    }
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return moviePages.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "suggestionTableViewCell", for: indexPath) as! SuggestionTableViewCell
+        
+        cell.setUp(moviePage: moviePages[indexPath.row])
+        
+        return cell
+    }
+    
+    private func searchMoviePage(sort: Sort, order: Order, label: String, isBestMovieHere: Bool){
+        _ = MovieService.shared().getMoviePage(sort: sort, order: order){ newMoviePage, response, error in
+            var moviePage = newMoviePage
+            moviePage.label = label
+            self.moviePages.append(moviePage)
+            DispatchQueue.main.async(){
+                if(isBestMovieHere){
+                    if let bestMovie = moviePage.results.first{
+                        _ = MovieService.shared().getMovieDetail(id: bestMovie.id!){ movie, response, error in
+                            self.setBestMovie(movie: movie)
```

Comment:
Take sure that use variables to service functions, if you is using this is because something is wrong, the services calls should be just methods and them don't have to be attributed to variables.